mtgjson5/price_builder.py

```diff
@@ -252,3 +297,159 @@ def build_prices(self) -> Tuple[Dict[str, Any], Dict[str, Any]]:
+                # Reconstruct MtgjsonPricesRecordV2 from dict
+                record = MtgjsonPricesRecordV2(
+                    provider=record_dict.get("provider", provider),
+                    treatment=record_dict["treatment"],
+                    currency=record_dict["currency"],
+                    price_value=record_dict["priceValue"],
+                    price_variant=record_dict["priceVariant"],
+                    uuid=record_dict["uuid"],
+                    platform=record_dict["platform"],
+                    price_type=record_dict["priceType"],
+                    date=record_dict["date"],
+                    subtype=record_dict.get("subtype"),
+                )
+                container.add_record(record)
+
+        return container
+
+    def _merge_v2_prices(
+        self,
+        archive: MtgjsonPricesV2Container,
+        today: MtgjsonPricesV2Container,
+    ) -> None:
+        """
+        Merge today's v2 price records into the archive.
+        Adds all of today's records to the archive (records are timestamped).
+
+        :param archive: Archive container to merge into
+        :param today: Today's prices to merge from
+        """
+        # Simply add all today's records to archive
+        # Each record is timestamped with today's date, so we preserve history
+        # pylint: disable=import-outside-toplevel
+        from .classes.mtgjson_prices_v2 import MtgjsonPricesRecordV2
+
+        today_serialized = today.to_json()
+        for _, records_list in today_serialized.items():
+            for record_dict in records_list:
+                # Reconstruct record and add to archive
+                record = MtgjsonPricesRecordV2(
+                    provider=record_dict["provider"],
+                    treatment=record_dict["treatment"],
+                    currency=record_dict["currency"],
+                    price_value=record_dict["priceValue"],
+                    price_variant=record_dict["priceVariant"],
+                    uuid=record_dict["uuid"],
+                    platform=record_dict["platform"],
+                    price_type=record_dict["priceType"],
+                    date=record_dict["date"],
+                    subtype=record_dict.get("subtype"),
+                )
+                archive.add_record(record)
+
+    def _prune_v2_prices_archive(self, archive: MtgjsonPricesV2Container) -> None:
+        """
+        Prune old v2 price records similar to legacy pruning.
+        Removes records older than configured cutoff date.
+
+        :param archive: Archive container to prune
+        """
+        LOGGER.info("Pruning v2 archive data")
+        cutoff_date = datetime.date.today() - datetime.timedelta(
+            weeks=int(MtgjsonConfig().get("Prices", "max_weeks", fallback="104"))
+        )
+        cutoff_date_str = cutoff_date.isoformat()
+
+        # Build a new container with only non-expired records
+        pruned_container = MtgjsonPricesV2Container()
```

Comment:
This is an interesting way of pruning data, Would delete record make more sense vs the complete replacement?